src/cacheHttpClient.ts

```diff
@@ -77,11 +81,25 @@ function createHttpClient(): HttpClient {
     );
 }
 
-export async function getCacheEntry(
-    keys: string[]
-): Promise<ArtifactCacheEntry | null> {
+export function getCacheVersion(): string {
+    // Add salt to cache version to support breaking changes in cache entry
+    const components = [
+        core.getInput(Inputs.Key, { required: true }),
+        core.getInput(Inputs.RestoreKeys, { required: false }),
+        core.getInput(Inputs.Path, { required: true }),
+        versionSalt
+    ];
+
+    return crypto
+        .createHash("sha256")
+        .update(components.join("|"))
+        .digest("hex");
+}
+
+export async function getCacheEntry(): Promise<ArtifactCacheEntry | null> {
     const httpClient = createHttpClient();
-    const resource = `cache?keys=${encodeURIComponent(keys.join(","))}`;
+    const version = getCacheVersion();
+    const resource = `cache?version=${version}`;
```

Comment:
@dhadka I took a stab at using a version to represent a cache. I'm not entirely sure how the endpoints expose this parameter, so I'm happy to revise as needed 😄

Reply:
We will still need to provide the keys here.  It should be `cache?keys=key1,key2,...,keyn&version=...`.

I also don't think we need to include Key and RestoreKeys in the version.  With the version, we only match caches with the exact `(key, version)` tuple.  So having the keys included in the version is redundant.

Reply:
Got it - updated to not include the keys in the `version` and include the `keys` as url params.

Reply:
@dhadka Why do we not need keys here?

I would think you could have an exact match on version but a restore key partial match since version only takes into account the action version and the input paths